diff --git a/config/Config-kernel.in b/config/Config-kernel.in
index c1bc7984f7..36194d42ab 100644
--- a/config/Config-kernel.in
+++ b/config/Config-kernel.in
@@ -1045,6 +1045,12 @@ if KERNEL_IP_MROUTE
 
 endif
 
+config KERNEL_MPTCP
+	bool "Enable IPv4 MultiPath TCP support"
+	default n
+	help
+	  IPv4 MultiPath TCP to kernel support.
+
 #
 # IPv6 configuration
 #
@@ -1086,6 +1092,13 @@ if KERNEL_IPV6
 	config KERNEL_LWTUNNEL_BPF
 		def_bool n
 
+	config KERNEL_MPTCP_IPV6
+		bool "Enable IPv6 MultiPath TCP support"
+		default n
+		depends on KERNEL_MPTCP
+		help
+		  IPv6 MultiPath TCP to kernel support.
+
 endif
 
 #
diff --git a/package/kernel/linux/modules/netsupport.mk b/package/kernel/linux/modules/netsupport.mk
index 27268ee043..f944968d97 100644
--- a/package/kernel/linux/modules/netsupport.mk
+++ b/package/kernel/linux/modules/netsupport.mk
@@ -1459,6 +1459,23 @@ endef
 $(eval $(call KernelPackage,xdp-sockets-diag))
 
 
+define KernelPackage/inet-mptcp-diag
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=INET diag support for MultiPath TCP
+  DEPENDS:= +@KERNEL_MPTCP +@KERNEL_MPTCP_IPV6 +kmod-inet-diag
+  KCONFIG:= CONFIG_INET_MPTCP_DIAG@ge5.6
+  FILES:= $(LINUX_DIR)/net/mptcp/mptcp_diag.ko@ge5.6
+  AUTOLOAD:=$(call AutoProbe,mptcp_diag)
+endef
+
+define KernelPackage/inet-mptcp-diag/description
+Support for INET (MultiPath TCP) socket monitoring interface used by
+native Linux tools such as ss.
+endef
+
+$(eval $(call KernelPackage,inet-mptcp-diag))
+
+
 define KernelPackage/wireguard
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
   TITLE:=WireGuard secure network tunnel
