diff --git a/net/ntpd/Makefile b/net/ntpd/Makefile
index 51a78403b..3e5800a31 100644
--- a/net/ntpd/Makefile
+++ b/net/ntpd/Makefile
@@ -8,19 +8,19 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ntp
-PKG_VERSION:=4.2.8p17
+PKG_VERSION:=4.2.8p18
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://www.eecis.udel.edu/~ntp/ntp_spool/ntp4/ntp-4.2/
-PKG_HASH:=103dd272e6a66c5b8df07dce5e9a02555fcd6f1397bdfb782237328e89d3a866
+PKG_HASH:=cf84c5f3fb1a295284942624d823fffa634144e096cfc4f9969ac98ef5f468e5
 
 PKG_LICENSE:=NTP
 PKG_LICENSE_FILES:=COPYRIGHT html/copyright.html
 PKG_CPE_ID:=cpe:/a:ntp:ntp
 
-PKG_FIXUP:=autoreconf
-PKG_LIBTOOL_PATHS:=. sntp
+#PKG_FIXUP:=autoreconf
+#PKG_LIBTOOL_PATHS:=. sntp
 PKG_CHECK_FORMAT_SECURITY:=0
 PKG_BUILD_PARALLEL:=1
 
diff --git a/net/samba4/Makefile b/net/samba4/Makefile
index 4afa87463..87a389455 100644
--- a/net/samba4/Makefile
+++ b/net/samba4/Makefile
@@ -39,6 +39,7 @@ PKG_CONFIG_DEPENDS:= \
 PYTHON3_PKG_BUILD:=0
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
 include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/kernel.mk
 include $(INCLUDE_DIR)/version.mk
@@ -68,7 +69,8 @@ define Package/samba4-libs
 	+PACKAGE_libpam:libpam \
 	+SAMBA4_SERVER_VFS:attr \
 	+SAMBA4_SERVER_AVAHI:libavahi-client \
-	+SAMBA4_SERVER_AD_DC:python3-cryptodome +SAMBA4_SERVER_AD_DC:python3-markdown +SAMBA4_SERVER_AD_DC:python3-dns +SAMBA4_SERVER_AD_DC:libopenldap +SAMBA4_SERVER_AD_DC:jansson +SAMBA4_SERVER_AD_DC:libarchive +SAMBA4_SERVER_AD_DC:acl +SAMBA4_SERVER_AD_DC:attr
+	+SAMBA4_SERVER_AD_DC:python3-cryptodome +SAMBA4_SERVER_AD_DC:python3-markdown +SAMBA4_SERVER_AD_DC:python3-dns +SAMBA4_SERVER_AD_DC:libopenldap +SAMBA4_SERVER_AD_DC:jansson +SAMBA4_SERVER_AD_DC:libarchive +SAMBA4_SERVER_AD_DC:acl +SAMBA4_SERVER_AD_DC:attr \
+	$(ICONV_DEPENDS)
 endef
 
 define Package/samba4-server
@@ -148,7 +150,7 @@ CONFIGURE_ARGS:=$(filter-out	\
 	--build=%			\
 	--program-prefix=%	\
 	--program-suffix=% 	\
-	--disable-nls		\
+	--enable-nls		\
 	--disable-ipv6		\
 	, $(CONFIGURE_ARGS))
 
@@ -178,7 +180,8 @@ CONFIGURE_ARGS += \
 		--disable-spotlight \
 		--enable-fhs \
 		--without-automount \
-		--without-iconv \
+		--with-libiconv="$(ICONV_PREFIX)" \
+		--with-iconv \
 		--without-lttng \
 		--without-pam \
 		--without-systemd \
