diff --git a/net/samba4/Makefile b/net/samba4/Makefile
index 4afa874..87a3894 100644
--- a/net/samba4/Makefile
+++ b/net/samba4/Makefile
@@ -39,6 +39,7 @@ PKG_CONFIG_DEPENDS:= \
 PYTHON3_PKG_BUILD:=0
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
 include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/kernel.mk
 include $(INCLUDE_DIR)/version.mk
@@ -68,7 +69,8 @@ define Package/samba4-libs
 	+PACKAGE_libpam:libpam \
 	+SAMBA4_SERVER_VFS:attr \
 	+SAMBA4_SERVER_AVAHI:libavahi-client \
-	+SAMBA4_SERVER_AD_DC:python3-cryptodome +SAMBA4_SERVER_AD_DC:python3-markdown +SAMBA4_SERVER_AD_DC:python3-dns +SAMBA4_SERVER_AD_DC:libopenldap +SAMBA4_SERVER_AD_DC:jansson +SAMBA4_SERVER_AD_DC:libarchive +SAMBA4_SERVER_AD_DC:acl +SAMBA4_SERVER_AD_DC:attr
+	+SAMBA4_SERVER_AD_DC:python3-cryptodome +SAMBA4_SERVER_AD_DC:python3-markdown +SAMBA4_SERVER_AD_DC:python3-dns +SAMBA4_SERVER_AD_DC:libopenldap +SAMBA4_SERVER_AD_DC:jansson +SAMBA4_SERVER_AD_DC:libarchive +SAMBA4_SERVER_AD_DC:acl +SAMBA4_SERVER_AD_DC:attr \
+	$(ICONV_DEPENDS)
 endef
 
 define Package/samba4-server
@@ -148,7 +150,7 @@ CONFIGURE_ARGS:=$(filter-out	\
 	--build=%			\
 	--program-prefix=%	\
 	--program-suffix=% 	\
-	--disable-nls		\
+	--enable-nls		\
 	--disable-ipv6		\
 	, $(CONFIGURE_ARGS))
 
@@ -178,7 +180,8 @@ CONFIGURE_ARGS += \
 		--disable-spotlight \
 		--enable-fhs \
 		--without-automount \
-		--without-iconv \
+		--with-libiconv="$(ICONV_PREFIX)" \
+		--with-iconv \
 		--without-lttng \
 		--without-pam \
 		--without-systemd \
